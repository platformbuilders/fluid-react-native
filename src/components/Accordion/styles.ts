import styled from 'styled-components/native';
import { moderateScale } from 'react-native-size-matters';
import DefaultIcon from '../Icon';

export const Icon = styled(DefaultIcon).attrs({
  touchable: false,
  size: moderateScale(24),
})``;
