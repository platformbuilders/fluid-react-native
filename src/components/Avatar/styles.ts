import styled from 'styled-components/native';
import { RNCamera } from 'react-native-camera';
import Touchable from '../Touchable';
import Icon from '../Icon';
import { getTheme, ifStyle } from '../../utils/helpers';

const accentMain = getTheme('accent.main');
const accentContrast = getTheme('accent.contrast');

const showBorder = ifStyle('showBorder');
const hasBorderWidth = ifStyle('borderWidth');
const hasBorderColor = ifStyle('borderColor');

type WrapperProps = {
  size: number;
  disabled: boolean;
  showBorder: boolean;
  borderWidth?: number;
  borderColor?: string;
};

export const Wrapper = styled(Touchable)<WrapperProps>`
  width: ${(props: WrapperProps): number => props.size}px;
  height: ${(props: WrapperProps): number => props.size}px;
  border-radius: ${(props: WrapperProps): number => props.size / 2}px;
  overflow: hidden;
  border-width: ${(props: WrapperProps): any =>
    showBorder(hasBorderWidth(props.borderWidth, '2'), '0')}px;
  border-color: ${(props: WrapperProps): any =>
    hasBorderColor(props.borderColor, accentMain)};
  position: relative;
`;

type IconWrapperProps = {
  size: number;
};

export const UploadIconWrapper = styled.View<IconWrapperProps>`
  align-items: center;
  justify-content: center;
  width: ${(props: IconWrapperProps): number => props.size / 4}px;
  height: ${(props: IconWrapperProps): number => props.size / 4}px;
  border-radius: ${(props: IconWrapperProps): number => props.size / 8}px;
  position: absolute;
  z-index: 2;
  background-color: ${accentMain};
`;

export const UploadIcon = styled(Icon).attrs({ name: 'camera' })`
  color: ${accentContrast};
`;

export const CameraView = styled(RNCamera)<{ size: number }>`
  width: ${(props: any): number => props.size}px;
  height: 100%;
  border-radius: ${(props: any): number => props.size / 2}px;
  overflow: hidden;
  border: ${showBorder('4px solid white', '')};
`;
