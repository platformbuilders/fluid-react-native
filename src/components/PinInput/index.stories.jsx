import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { withKnobs } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import PinInput from './index';
import PinInputMarkdown from './pinInput.md';

storiesOf('PinInput', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: PinInputMarkdown,
      sidebar: PinInputMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Simple example', () => (
    <PinInput onChangeText={action('PinInput-change-text')} />
  ));
