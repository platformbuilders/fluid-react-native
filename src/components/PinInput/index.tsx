import React, { useContext, useState } from 'react';
import { moderateScale } from 'react-native-size-matters';
import { ThemeContext } from 'styled-components';
import { PinInputType } from '../../types';
import FormError from '../FormError';
import {
  Wrapper,
  PinCodeInput,
  Icon,
  CaptionText,
  defaultStyling,
} from './styles';

const iconString = (hidePassword: boolean): string =>
  hidePassword ? 'eye' : 'eye-off';

const PinInput: React.FC<PinInputType> = ({
  password = false,
  animated = false,
  centered = false,
  contrast = false,
  mask = 'â€¢',
  codeLength = 4,
  cellSpacing = moderateScale(6),
  iconSize = moderateScale(24),
  cellSize = moderateScale(56),
  caption,
  error,
  value,
  onChangeText,
  onFulfill = () => {},
  wrapperStyle,
  ...rest
}) => {
  const [hidePassword, setHidePassword] = useState(true);
  const theme = useContext(ThemeContext);
  const defaultStyle = defaultStyling(theme);
  const changeText = (text: string) => {
    onChangeText(text);
    if (text.length === codeLength) {
      onFulfill(text);
    }
  };

  return (
    <FormError centered={centered} error={error}>
      <Wrapper style={wrapperStyle}>
        <PinCodeInput
          value={value}
          onTextChange={changeText}
          password={password && hidePassword}
          mask={mask}
          codeLength={codeLength}
          animated={animated}
          cellSize={cellSize}
          cellSpacing={cellSpacing}
          {...defaultStyle}
          {...rest}
        />
        {password && (
          <Icon
            accessibility={`Exibir ou ocultar inputs - ${iconString(
              hidePassword,
            )}`}
            name={iconString(hidePassword)}
            size={iconSize}
            onPress={() => setHidePassword(!hidePassword)}
            contrast={contrast}
            error={false}
          />
        )}
      </Wrapper>
      {!!caption && (
        <CaptionText accessibility={caption} centered={centered}>
          {caption}
        </CaptionText>
      )}
    </FormError>
  );
};

export default PinInput;
