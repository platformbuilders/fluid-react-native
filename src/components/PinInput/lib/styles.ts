import { StyleSheet } from 'react-native';

export const DefaultStyles = StyleSheet.create({
  containerDefault: {},
  cellDefault: {
    borderColor: 'gray',
    borderWidth: 1,
  },
  cellFocusedDefault: {
    borderColor: 'black',
    borderWidth: 2,
  },
  textStyleDefault: {
    color: 'gray',
    fontSize: 24,
  },
  textStyleFocusedDefault: {
    color: 'black',
  },
});
