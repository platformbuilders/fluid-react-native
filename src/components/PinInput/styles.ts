import styled from 'styled-components/native';
import DefaultCodeInput from 'react-native-smooth-pincode-input';
import { moderateScale } from 'react-native-size-matters';
import { ThemeType } from '../../types';
import { ifStyle, getTheme } from '../../utils/helpers';
import Typography from '../Typography';
import DefaultIcon from '../Icon';

const hasError = ifStyle('error');
const isContrast = ifStyle('contrast');
const isCentered = ifStyle('centered');
const failure = getTheme('failure');
const primaryMain = getTheme('primary.main');
const disabledDark = getTheme('primary.Dark');
const primaryContrast = getTheme('primary.contrast');
const mediumSpacing = getTheme('mediumSpacing');

type WrapperProps = {
  style: any;
};

export const defaultStyling = (theme?: ThemeType): any => ({
  cellStyle: {
    borderRadius: moderateScale(8),
    borderColor: theme ? disabledDark(theme) : '#eeeeee',
    borderWidth: 2,
  },
  cellStyleFocused: {
    borderColor: theme ? primaryMain(theme) : '#eeeeee',
  },
  textStyle: {
    fontSize: moderateScale(26),
    color: 'black',
  },
});

export const Wrapper = styled.View<WrapperProps>`
  align-items: center;
  flex-direction: row;
`;

export const PinCodeInput = styled(DefaultCodeInput)``;

type IconProps = {
  contrast: boolean;
  error: boolean;
};
export const Icon = styled(DefaultIcon).attrs((props: IconProps) => ({
  color: hasError(
    failure(props),
    isContrast(primaryContrast(props), primaryMain(props))(props),
  )(props),
}))<IconProps>`
  margin-left: ${mediumSpacing};
`;

type CaptionProps = {
  centered: boolean;
};

export const CaptionText = styled(Typography).attrs({
  variant: 'footnote',
})<CaptionProps>`
  text-align: ${isCentered('center', 'left')};
  opacity: 0.67;
  font-weight: 300;
`;
