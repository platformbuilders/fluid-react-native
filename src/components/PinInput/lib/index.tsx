/* eslint-disable sonarjs/cognitive-complexity */
/* eslint-disable prefer-spread */
import React, { useEffect, useState } from 'react';
import {
  I18nManager,
  ImageStyle,
  KeyboardTypeOptions,
  NativeSyntheticEvent,
  Text,
  TextInput,
  TextInputKeyPressEventData,
  TextInputProps,
  TextProps,
  TextStyle,
  View,
  ViewProps,
  ViewStyle,
} from 'react-native';
import * as Animatable from 'react-native-animatable';
import { DefaultStyles } from './styles';

type Props = {
  value?: string;
  codeLength?: number;
  cellSize?: number;
  cellSpacing?: number;
  placeholder?: string | React.FC;
  mask?: string | React.FC;
  maskDelay?: number;
  password?: boolean;
  autoFocus?: boolean;
  restrictToNumbers?: boolean;
  containerStyle?: ViewProps;
  cellStyle?: ViewProps;
  cellStyleFocused?: ViewProps;
  cellStyleFilled?: ViewProps;
  textStyle?: TextProps;
  textStyleFocused?: TextProps;
  animated?: boolean;
  animationFocused?:
    | string
    | Animatable.CustomAnimation<TextStyle & ViewStyle & ImageStyle>;
  onFulfill?(code: string): void;
  onChangeText?(code: string): void;
  onBackspace?(): void;
  onTextChange?(code: string): void;
  testID?: string;
  onFocus?(): void;
  onBlur?(): void;
  keyboardType?: KeyboardTypeOptions;
  editable?: boolean;
  disableFullscreenUI?: boolean;
  inputProps?: TextInputProps;
  useAccessibilityLabelWithIndex?: string;
  useIndexedTestID?: string;
};

const SmoothPinCodeInput: React.FC<Props> = ({
  value = '',
  codeLength = 4,
  cellSize = 48,
  cellSpacing = 4,
  placeholder = '',
  password = false,
  mask = '*',
  maskDelay = 200,
  keyboardType = 'numeric',
  autoFocus = false,
  restrictToNumbers = false,
  containerStyle = DefaultStyles.containerDefault,
  cellStyle = DefaultStyles.cellDefault,
  cellStyleFocused = DefaultStyles.cellFocusedDefault,
  textStyle = DefaultStyles.textStyleDefault,
  textStyleFocused = DefaultStyles.textStyleFocusedDefault,
  animationFocused = 'pulse',
  animated = true,
  editable = true,
  inputProps = {},
  disableFullscreenUI = true,
  useAccessibilityLabelWithIndex = undefined,
  useIndexedTestID = undefined,
  cellStyleFilled = {},
  onBackspace,
  onBlur,
  onFocus,
  onFulfill,
  onTextChange,
  testID,
}) => {
  const [maskTimeout, setMaskTimeout] = useState<NodeJS.Timeout>();
  const [maskDelayLocal, setMaskDelayLocal] = useState(false);
  const [focused, setFocused] = useState(false);

  const ref = React.createRef<TextInput>();

  const inputRef = React.createRef<TextInput>();

  const _inputCode = (code: string) => {
    // const { password, codeLength = 4, onTextChange, onFulfill } = props;
    if (restrictToNumbers) {
      code = (code.match(/[0-9]/g) || []).join('');
    }
    if (onTextChange) {
      onTextChange(code);
    }
    if (code.length === codeLength && onFulfill) {
      onFulfill(code);
    }
    // handle password mask
    const maskDelayLocalConst = password && code.length > value.length; // only when input new char
    setMaskDelayLocal(maskDelayLocalConst);
    if (maskDelayLocal) {
      // mask password after delay
      if (maskTimeout) clearTimeout(maskTimeout);
      setMaskTimeout(
        setTimeout(() => {
          setMaskDelayLocal(false);
        }, maskDelay),
      );
    }
  };
  const _keyPress = (
    event: NativeSyntheticEvent<TextInputKeyPressEventData>,
  ) => {
    if (event.nativeEvent.key === 'Backspace') {
      // const { value, onBackspace } = props;
      if (value === '' && onBackspace) {
        onBackspace();
      }
    }
  };
  const _onFocused = () => {
    setFocused(true);
    if (typeof onFocus === 'function') {
      onFocus();
    }
  };
  const _onBlurred = () => {
    setFocused(false);
    if (typeof onBlur === 'function') {
      onBlur();
    }
  };
  const useIndexedAccessibilityLabelLocal = (idx: number) => {
    if (typeof useAccessibilityLabelWithIndex === 'string')
      return `${useAccessibilityLabelWithIndex} ${idx}`;
    return `Insira o PIN ${idx}`;
  };
  const useIndexedTestIDLocal = (idx: number) => {
    if (typeof useIndexedTestID === 'string')
      return `${useIndexedTestID} ${idx}`;
    return `Insira o PIN ${idx}`;
  };

  useEffect(() => {
    return () => {
      if (maskTimeout) clearTimeout(maskTimeout);
    };
  }, []);

  return (
    <Animatable.View
      ref={ref}
      style={[
        {
          alignItems: 'stretch',
          flexDirection: 'row',
          justifyContent: 'center',
          position: 'relative',
          width: cellSize * codeLength + cellSpacing * (codeLength - 1),
          height: cellSize,
        },
        containerStyle,
      ]}
    >
      <View
        style={{
          position: 'absolute',
          margin: 0,
          height: '100%',
          flexDirection: I18nManager.isRTL ? 'row-reverse' : 'row',
          alignItems: 'center',
        }}
      >
        {Array.apply(null, Array(codeLength)).map((_, idx) => {
          const cellFocused = focused && idx === value.length;
          const filled = idx < value.length;
          const last = idx === value.length - 1;
          const showMask = filled && password && (!maskDelayLocal || !last);
          const isPlaceholderText = typeof placeholder === 'string';
          const isMaskText = typeof mask === 'string';
          const pinCodeChar = value.charAt(idx);
          let cellText = null;
          if (filled || placeholder !== null) {
            if (showMask && isMaskText) {
              cellText = mask;
            } else if (!filled && isPlaceholderText) {
              cellText = placeholder;
            } else if (pinCodeChar) {
              cellText = pinCodeChar;
            }
          }
          const placeholderComponent = !isPlaceholderText ? placeholder : null;
          const maskComponent = showMask && !isMaskText ? mask : null;
          const isCellText = typeof cellText === 'string';
          return (
            <Animatable.View
              key={idx}
              style={[
                {
                  width: cellSize,
                  height: cellSize,
                  marginLeft: cellSpacing / 2,
                  marginRight: cellSpacing / 2,
                  flexDirection: 'row',
                  alignItems: 'center',
                  justifyContent: 'center',
                },
                cellStyle,
                cellFocused ? cellStyleFocused : {},
                filled ? cellStyleFilled : {},
              ]}
              animation={
                idx === value.length && focused && animated
                  ? animationFocused
                  : undefined
              }
              iterationCount="infinite"
              duration={500}
              accessibilityLabel={useIndexedAccessibilityLabelLocal(idx)}
              testID={useIndexedTestIDLocal(idx)}
            >
              {isCellText && !maskComponent && (
                <Text style={[textStyle, cellFocused ? textStyleFocused : {}]}>
                  {cellText}
                </Text>
              )}
              {!isCellText && !maskComponent && placeholderComponent}
              {isCellText && maskComponent}
            </Animatable.View>
          );
        })}
      </View>
      <TextInput
        disableFullscreenUI={disableFullscreenUI}
        value={value}
        ref={inputRef}
        onChangeText={_inputCode}
        onKeyPress={_keyPress}
        onFocus={_onFocused}
        onBlur={_onBlurred}
        spellCheck={false}
        autoFocus={autoFocus}
        keyboardType={keyboardType}
        numberOfLines={1}
        caretHidden
        maxLength={codeLength}
        selection={{
          start: value.length,
          end: value.length,
        }}
        style={{
          flex: 1,
          opacity: 0,
          textAlign: 'center',
        }}
        testID={testID || undefined}
        editable={editable}
        {...inputProps}
      />
    </Animatable.View>
  );
};

export default SmoothPinCodeInput;

//   static defaultProps = {
//     value: '',
//     codeLength: 4,
//     cellSize: 48,
//     cellSpacing: 4,
//     placeholder: '',
//     password: false,
//     mask: '*',
//     maskDelay: 200,
//     keyboardType: 'numeric',
//     autoFocus: false,
//     restrictToNumbers: false,
//     containerStyle: styles.containerDefault,
//     cellStyle: styles.cellDefault,
//     cellStyleFocused: styles.cellFocusedDefault,
//     textStyle: styles.textStyleDefault,
//     textStyleFocused: styles.textStyleFocusedDefault,
//     animationFocused: 'pulse',
//     animated: true,
//     editable: true,
//     inputProps: {},
//     disableFullscreenUI: true,
//     useAccessibilityLabelWithIndex: undefined,
//     useIndexedTestID: undefined,
//   };
// }
// SmoothPinCodeInput.propTypes = {
//   value: PropTypes.string,
//   codeLength: PropTypes.number,
//   cellSize: PropTypes.number,
//   cellSpacing: PropTypes.number,
//   placeholder: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
//   mask: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
//   maskDelay: PropTypes.number,
//   password: PropTypes.bool,
//   autoFocus: PropTypes.bool,
//   restrictToNumbers: PropTypes.bool,
//   containerStyle: ViewPropTypes.style,
//   cellStyle: ViewPropTypes.style,
//   cellStyleFocused: ViewPropTypes.style,
//   cellStyleFilled: ViewPropTypes.style,
//   textStyle: Text.propTypes.style,
//   textStyleFocused: Text.propTypes.style,
//   animated: PropTypes.bool,
//   animationFocused: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
//   onFulfill: PropTypes.func,
//   onChangeText: PropTypes.func,
//   onBackspace: PropTypes.func,
//   onTextChange: PropTypes.func,
//   testID: PropTypes.any,
//   onFocus: PropTypes.func,
//   onBlur: PropTypes.func,
//   keyboardType: PropTypes.string,
//   editable: PropTypes.bool,
//   inputProps: PropTypes.exact(TextInput.propTypes),
//   useAccessibilityLabelWithIndex: PropTypes.string,
//   useIndexedTestID: PropTypes.string,
