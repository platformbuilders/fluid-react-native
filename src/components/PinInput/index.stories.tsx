import React, { useState } from 'react';
import type { Meta } from '@storybook/react';
import PinInput from '.';

const RenderPinInput = ({ ...args }) => {
  const [value, setValue] = useState('');

  const onChangeText = (text: string) => {
    setValue(text);
  };
  return <PinInput {...args} value={value} onChangeText={onChangeText} />;
};

const meta: Meta<typeof PinInput> = {
  title: 'PinInput',
  component: RenderPinInput,
  args: {
    accessibility: 'Digite seu pin',
    animated: true,
    autoFocus: true,
    borderedLabel: 'Label fica aqui',
    caption: 'Caption fica aqui',
    cellSize: 40,
    cellSpacing: 10,
    cellStyle: { backgroundColor: '#193f5e', borderRadius: 10 },
    cellStyleFocused: { backgroundColor: '#dbd6f3', borderRadius: 10 },
    centered: true,
    codeLength: 4,
    contrast: true,
    editable: true,
    error: 'Mensagem de erro aqui.',
    fixedLabelVariant: 'md',
    id: 'pin',
    inputProps: { keyboardType: 'numeric' },
    keyboardAppearance: 'dark',
    keyboardType: 'numeric',
    labelStyle: { color: '#193f5e' },
    onBackspace: () => console.log('Backspace'),
    onFulfill: (code: string) => console.log(code),
    placeholder: '0',
  },
  parameters: {
    notes: 'Este componente Ã© utilizado para renderizar um input de pin.',
  },
};

export default meta;

export const Default = {};

export const Password: Meta<typeof PinInput> = {
  args: {
    ...meta.args,
    inputProps: { keyboardType: 'numeric' },
    keyboardType: 'numeric',
    password: true,
    iconSize: 24,
  },
};
