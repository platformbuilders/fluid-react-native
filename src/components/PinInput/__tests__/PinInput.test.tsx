import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import PinInput from '..';
import { defaultStyling } from '../styles';
import { theme } from '../../../test/helpers';

describe('<PinInput />', () => {
  it('should render textinput', () => {
    const onChangeText = jest.fn();
    const defaultStyle = defaultStyling(theme);
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <PinInput
          id="test"
          accessibility=""
          onChangeText={onChangeText}
          {...defaultStyle}
        />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
