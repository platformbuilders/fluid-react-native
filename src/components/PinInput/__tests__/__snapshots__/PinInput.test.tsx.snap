// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PinInput /> should render textinput 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 18.857142857142858,
      },
    ]
  }
>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "row",
        "height": 88,
        "justifyContent": "center",
        "position": "relative",
        "width": 380.2857142857143,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "100%",
          "margin": 0,
          "position": "absolute",
        }
      }
    >
      <View
        forwardedRef={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": undefined,
            "borderRadius": 12.571428571428571,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 88,
            "justifyContent": "center",
            "marginLeft": 4.714285714285714,
            "marginRight": 4.714285714285714,
            "width": 88,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 40.85714285714286,
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        forwardedRef={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": undefined,
            "borderRadius": 12.571428571428571,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 88,
            "justifyContent": "center",
            "marginLeft": 4.714285714285714,
            "marginRight": 4.714285714285714,
            "width": 88,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 40.85714285714286,
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        forwardedRef={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": undefined,
            "borderRadius": 12.571428571428571,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 88,
            "justifyContent": "center",
            "marginLeft": 4.714285714285714,
            "marginRight": 4.714285714285714,
            "width": 88,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 40.85714285714286,
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        forwardedRef={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": undefined,
            "borderRadius": 12.571428571428571,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 88,
            "justifyContent": "center",
            "marginLeft": 4.714285714285714,
            "marginRight": 4.714285714285714,
            "width": 88,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 40.85714285714286,
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      caretHidden={true}
      disableFullscreenUI={true}
      editable={true}
      keyboardType="numeric"
      maxLength={4}
      numberOfLines={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      selection={
        Object {
          "end": 0,
          "start": 0,
        }
      }
      spellCheck={false}
      style={
        Object {
          "flex": 1,
          "opacity": 0,
          "textAlign": "center",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
